lib/bambora/v1/payment_resource.rb

```diff
@@ -3,15 +3,24 @@
 module Bambora
   module V1
     class PaymentResource
-      # Summary: Create and modify payments.
-      # Note: The link below links to all apis includding profiles and tokenization. There aren't great docs explaining
-      #       the /payments endpoints alone.
-      # Docs: https://dev.na.bambora.com/docs/references/payment_APIs/
-      #       https://dev.na.bambora.com/docs/references/payment_SDKs/take_payments/?shell#
-      # Endpoint: https://api.na.bambora.com/v1/payments
-      def initialize(client:, sub_path:)
+      def initialize(client:)
         @client = client
-        @sub_path = sub_path
+        @sub_path = '/v1/payments'
+      end
+
+      def make_payment(payment_data)
+        @client.request(method: :post, path: @sub_path, body: payment_data)
```

Comment:
Lower-Level API method.

Reply:
How does this handle failures? Failure on the network? Failure with the transaction?

Reply:
@erawk handling failures like this should be happening at the client level (`Bambora::JSONClient`). You can see that we do [*some* error-handling](https://github.com/HiMamaInc/bambora-client/blob/master/lib/bambora/json_client.rb#L15), but this could be more robust and `rescue` from more errors, such as ones that would result from the issues you mentioned above.

Comparing the error handling done in our gem versus the `beanstream` gem, it is possible to see that there is in fact [error handling for some of the errors you mentioned](https://github.com/bambora-na/beanstream-ruby/blob/master/lib/beanstream/transaction.rb#L87). I think this work is necessary moving forward, but I think is out of scope of this PR. I have created a [tile](https://trello.com/c/teSIlyMz) to address this.

As a safety precaution I searched BugSnag for some of the text that was contained in the `beanstream` gem REST client error raising I linked to above, and could not find a single record that any of these errors had occurred. To an end-user, I imagine this would appear as a 500 error, at which point they would have to go back, and re-input the action.

Given the low probability of this happening, given my searching through BugSnag, I feel it is okay to push forward, but prioritize this failure handling.

Reply:
Yes, follow up tile makes sense. I think the client should handle some category of errors, like network, server communication and TLS issues. Others ought to be handled by the consumer of the client as you suggest.